### PR TITLE
DAOS-13232 bio: adjust minmal hugepages requirement

### PR DESCRIPTION
**Pull request by NiuYawei**
Reduced the minimal per target hugepages requirement from 1GB to 256MB,
changed initial allocation from 256MB to 192MB.

Remove hard validation checks if number of hugepages specified in the
config file is not able to be allocated. Control-plane will continue
to auto-calculate and assign number of huge pages in server config
file based on 1gib per engine target to satisfy DMA buffer requirement
for average workload but if this requirement cannot be met then a
NOTICE log level message will be printed but the engine will still be
able to start. Memory requirement validation will be left to the
engine after process is spawned. This change allows engine to be
started with lower number of hugepages allocated in resource
constrained environments. When there are insufficient free hugepages
to satisfy the number requested in config, engine mem_size parameter
is calculated based on the free number of hugepages rather than the
number requested in config.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by NiuYawei** [comment]
@tanabarr Is there any restraint on control plane side needs be adjusted accordingly?

File: src/bio/bio_xstream.c [hunk 1]
```diff
@@ -31,9 +31,9 @@
 #define DAOS_BS_CLUSTER_SZ	(1ULL << 30)	/* 1GB */
 /* DMA buffer parameters */
 #define DAOS_DMA_CHUNK_MB	8	/* 8MB DMA chunks */
-#define DAOS_DMA_CHUNK_CNT_INIT	32	/* Per-xstream init chunks */
-#define DAOS_DMA_CHUNK_CNT_MAX	128	/* Per-xstream max chunks */
-#define DAOS_DMA_MIN_UB_BUF_MB	1024	/* 1GB min upper bound DMA buffer */
+#define DAOS_DMA_CHUNK_CNT_INIT	24	/* Per-xstream init chunks, 192MB */
+#define DAOS_DMA_CHUNK_CNT_MAX	128	/* Per-xstream max chunks, 1GB */
+#define DAOS_DMA_CHUNK_CNT_MIN	32	/* Per-xstream max chunks, 256MB */
```
Comment (wangshilong) [inline src/bio/bio_xstream.c:36]: [typo] this should be min chunks.

**Review by wangshilong — DISMISSED** [review]

**Review by tanabarr — DISMISSED** [review]

**Comment by tanabarr** [comment]
> @tanabarr Is there any restraint on control plane side needs be adjusted accordingly?

I don't think so. Presumably when we are calculating the number of hugepages for each engine we still want to be using the 1gib-per-target estimation to allow for average workload? We do check if this allocation of hugemem can be met and if not then we don't start the engine, should we relax that validation?

**Review by gnailzenh — APPROVED** [review]

**Review by wangshilong — DISMISSED** [review]

**Review by wangshilong — APPROVED** [review]

**Review by kjacque — APPROVED** [review]
Changes seem reasonable to me.

**Review by tanabarr — APPROVED** [review]

**Pull request merged**